pytket/extensions/qiskit/qiskit_convert.py

```diff
@@ -518,41 +520,49 @@ def _build_if_else_circuit(
     qubits: list[Qubit],
     bits: list[Bit],
 ) -> Circuit:
-    # Coniditions must be on a single bit (for now) TODO: support multiple bits.
-    if len(bits) == 1:
-        # Get two CircBox objects which implement the true_body and false_body.
-        if_box, else_box = _pytket_boxes_from_ifelseop(if_else_op, qregs, cregs)
-        # else_box can be None if no false_body is specified.
-        circ_builder = CircuitBuilder(qregs, cregs)
-        circ = circ_builder.circuit()
-    else:
-        raise NotImplementedError("Conditions over multiple bits not yet supported.")
-
-    # Coniditions must be on a single bit (for now)
-    if not isinstance(if_else_op.condition[0], Clbit):
-        raise NotImplementedError(
-            "Handling of register conditions is not yet supported"
+    # Get two CircBox objects which implement the true_body and false_body.
+    if_box, else_box = _pytket_boxes_from_ifelseop(if_else_op, qregs, cregs)
+    # else_box can be None if no false_body is specified.
+    circ_builder = CircuitBuilder(qregs, cregs)
+    circ = circ_builder.circuit()
+
+    if isinstance(if_else_op.condition[0], Clbit):
+        if len(bits) != 1:
+            raise NotImplementedError("Conditions on multiple bits not supported")
```

Comment:
Is there an issues / is this something we want to add in the future?

Reply:
Possibly but it may be more annoying to handle and maintain. I'm not certain on the limits on what you can express with `QuantumCircuit.if_test` (I should check).

I think for now we should stick with single bit conditions and entire register command conditions. If theres a demand for different cases then we can decide to support that.